components/model.py

```diff
@@ -926,6 +925,104 @@ def get_prerequisite(module_code):
     return DB_CURSOR.fetchall()
 
 
+def get_prerequisite_as_string(module_code):
+    '''
+        Returns a string of pre-requisites of specified module_code
+    '''
+    prerequisites = get_prerequisite(module_code)
+    prereq_list = convert_to_list(prerequisites)
+
+    # sort list of lists based on index (which is the first elem of each row)
+    prereq_list.sort(key=lambda row: row[INDEX_FIRST_ELEM])
+
+    prereq_string = convert_list_of_prereqs_to_string(prereq_list)
+
+    return prereq_string
+
+
+def convert_list_of_prereqs_to_string(prereq_list):
```

Comment:
Your algorithm is wrong.

Sample fail case:
**For module CP3880**

Expected:
[ IS2102 or CG1413 and (CS2103 or IS2103 or IS2150)] and preferably completed [IS3102 or (CS3201 and CS3202) or (CS3281 and CS3282) or IS4102 or (CS4201 and CS4202) or (CS4203 and CS4204) or (CG2001). Student Selection process will be enforced.
(my parser ignored all clauses after 'preferably completed ...' as that clause does not form the true prerequisites for this module.)

Actual:
![image](https://cloud.githubusercontent.com/assets/12386860/24079467/bb88ab3e-0cc3-11e7-98bb-e45b4aeac219.png)

Diff:

**Expected**: IS2102 or CG1413 ***and (CS2103 or IS2103 or IS2150)***
**Actual**: IS2102 or CG1413 ***or CS2103 or IS2103 or IS2150***